### ISSUE TITLE
createSymbolicLink: already exists (File exists)

### ISSUE THREAD

**Issue by koterpillar** [fpco/cache-s3#3]
I'm not sure whether this is the same (or has the same cause) as #2, but here's another error on downloading the same cache on a reused build agent:

```
[cache-s3][Info ][Tue, 30 Jan 2018 03:11:46 UTC]: <cache-s3/xxxxxxxxxxxxxx/master.lts-10.3.stack.cache> - Restoring cache with total size: 724.7 MiB
cache-s3: /home/xxxxxxxxx/.stack/programs/x86_64-linux/ghc-8.2.2/bin/ghc: createSymbolicLink: already exists (File exists)
```

See the relevant archive contents in #2.

**Comment by koterpillar** [comment]
Note: this is cache-s3 version 0.1.1a. As a workaround, I'm now deleting `~/.stack` before restoring the cache.

**Comment by lehins** [comment]
This was something I didn't expect. `System.Directory.doesFileExist` and `System.Posix.Files.fileExist` also follow symbolic links, so even if the link exists they will return `False` whenever link is broken. Good to know. @koterpillar I'll push a fix in a little bit and will let you when there is a working executable.

**Comment by lehins** [comment]
Fixed in [v0.1.2](https://github.com/fpco/cache-s3/releases/tag/v0.1.2)
